bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java

```diff
@@ -475,18 +475,37 @@ private boolean isSupportedFormat(final String format) {
    }
 
    @Override
-   public void subscribe(final String modelUri, final SubscriptionListener subscriptionListener, final String format) {
-      String checkedFormat = checkedFormat(format);
+   public void subscribe(final String modelUri, final SubscriptionListener subscriptionListener) {
+      Request request = new Request.Builder()
+         .url(
+            makeWsUrl(
+               createHttpUrlBuilder(makeUrl(SUBSCRIPTION))
+                  .addQueryParameter("modeluri", modelUri)
+                  .build()
+                  .toString()))
+         .build();
+
+      subscribe(modelUri, subscriptionListener, request);
+   }
+
+   @Override
+   public void subscribe(final String modelUri, final SubscriptionListener subscriptionListener,
+      final long timeout) {
       Request request = new Request.Builder()
          .url(
             makeWsUrl(
                createHttpUrlBuilder(makeUrl(SUBSCRIPTION))
                   .addQueryParameter("modeluri", modelUri)
-                  .addQueryParameter("format", checkedFormat)
+                  .addQueryParameter("timeout", String.valueOf(timeout))
                   .build()
                   .toString()))
          .build();
 
+      subscribe(modelUri, subscriptionListener, request);
+   }
+
+   private void subscribe(final String modelUri, final SubscriptionListener subscriptionListener,
```

Comment:
You should avoid using the same method name for public and private methods; it can confuse the compiler (Especially when the private method uses non-API types, such as `Request` here). You can rename this to e.g. doSubscribe()

Otherwise, client-applications get errors such as:

`The type okhttp3.Request cannot be resolved. It is indirectly referenced from required .class files`

Reply:
Thank you for your feedback Camille, I addressed this issue now!